ui/src/table.tsx

```diff
@@ -491,14 +491,39 @@ const
             } />
         )
       }, []),
-      onToggleCollapseAll = (isAllCollapsed: B) => expandedRefs.current = isAllCollapsed ? {} : null,
+      onToggleCollapseAll = (isAllCollapsed: B) => {
+        if (m.events?.includes('group_change')) {
+          let changingGroups 
+          if (isAllCollapsed){
+            if (expandedRefs.current && Object.keys(expandedRefs.current).length > 0){
+              changingGroups = Object.keys(expandedRefs.current)
+            } else {
+              changingGroups = groups?.map(group => group.name)
+            }
+          } else {
+            changingGroups = groups?.map(group => group.name)
+          }
```

Comment:
Would this work as well?

```suggestion
          const changedGroups =
            isAllCollapsed && expandedRefs.current && Object.keys(expandedRefs.current).length > 0
              ? Object.keys(expandedRefs.current)
              : groups?.map(group => group.name)
```

Reply:
Done. Thanks for the suggestion - I'm usually lean towards explicit if-else blocks, but what you have is definitely cleaner and more concise!